cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+        msg = Text('.//h3[contains(@id,"migration-plan-results-message")]')
+
+        @property
+        def is_displayed(self):
+            return self.msg.is_displayed
+
+    def after_fill(self, was_change):
+        self.close.click()
+
+    @property
+    def is_displayed(self):
+        return self.title.text == "Create Migration Plan"
+
+
+class MigrationPlanRequestDetailsView(View):
+    migration_request_details_list = MigrationPlanRequestDetailsList("plan-request-details-list")
+    sort_type = SelectorDropdown("id", "sortTypeMenu")
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    sort_order = Text(".//button[./span[contains(@class,'sort-direction')]]")
+    filter_by_dropdown = SelectorDropdown("id", "filterFieldTypeMenu")
+    filter_by_status_dropdown = SelectorDropdown("id", "filterCategoryMenu")
+
+    @property
+    def is_displayed(self):
+        return self.migration_request_details_list.is_displayed
+
+    def after_fill(self, was_change):
+        self.browser.send_keys(Keys.ENTER, self.search_box)
+
+    def filter_by_vm_name(self, vm_name):
+        """Enter VM Name in search box and hit ENTER to filter the list of VMs.
+
+        Args:
+            vm_name(str): Takes VM Name as arg.
+        """
+        try:
+            self.filter_by_dropdown.item_select("VM Name")
+        except NoSuchElementException:
+            logger.info(
+                "filter_by_dropdown not present, "
+                "migration plan may not have started yet.Ignoring."
+            )
+        self.search_box.fill(vm_name)
+        self.after_fill(was_change=True)
+
+    def get_migration_status_by_vm_name(self, vm_name):
+        """Search VM using filter_by_name and return its status.
+
+        Args:
+            vm_name(str): Takes VM Name as arg.
+        """
+        try:
+            # Try to clear previously applied filters, if any.
+            self.clear_filters.click()
+        except NoSuchElementException:
+            # Ignore as button won't be visible if there were no filters applied.
+            logger.info("Clear Filters button not present, ignoring.")
+        self.filter_by_vm_name(vm_name)
+        status = {
+            "Message": self.migration_request_details_list.get_message_text(vm_name),
+            "Description": self.migration_request_details_list.get_progress_description(vm_name),
+            "Time Elapsed": self.migration_request_details_list.get_clock(vm_name),
+        }
+        return status
+
+    def filter_by_status(self, status):
+        """Set filter_by_dropdown to 'Status' and uses status arg by user to set status filter.
+
+        Args:
+            status(str): Takes status string as arg. Valid status options are:
+                         ['Pending', 'Validating', 'Pre-migration', 'Migrating',
```

Comment:
Consider defining these as constants on the View class or on the entity for a MigrationPlan. Then we can consistently reference them without string typos, like in 'VM Transformations Ccompleted' below.